src/query/storage/m3/storage.go

```diff
@@ -174,80 +173,73 @@ func (s *m3storage) FetchTags(
 	query *storage.FetchQuery,
 	options *storage.FetchOptions,
 ) (*storage.SearchResults, error) {
+	tagResult, cleanup, err := s.SearchRaw(ctx, query, options)
+	if err != nil {
+		return nil, err
+	}
+
+	metrics := make(models.Metrics, len(tagResult))
+	for i, result := range tagResult {
+		m, err := storage.FromM3IdentToMetric(result.ID, result.Iter, s.tagOptions)
+		if err != nil {
+			return nil, err
+		}
+
+		metrics[i] = m
+	}
+
+	cleanup()
```

Comment:
Can you move this to a `defer cleanup()` before `metrics := make(model.Metrics...` so that it will get called in the case of an early return in the for loop?

Reply:
Good call